Converted to Polymer 2.0 (#7) :D

mikeperret opened the pull request:
Hi there

I made this work with Polymer 2.0:

- Changed emoji-rain.html to the new syntax
- emoji-rain.html: Added setTimeout to the resize method (works somehow more reliably)
- emoji-rain.html: "Speed" is now also an attribute/property
- Updated `iron-component-page` to 3.0.1 (added analysis.json)
- Not sure if necessary: I added polymer.json so it worked for me with `polymer serve` and `polymer test`

About this I am not sure: 
I also changed the paths of the HTML-Imports, as for me, the bower_components directory is in the same folder as index.html and emoji-rain.html. I hope I didn't mess anything up for you. But I think it should be fine when you can make a git clone, run `bower install `and then `polymer serve` and it works, correct? 

Also this is my first pull request ever, hope I didn't mess it up 😆

mikeperret committed:
Converted to Polymer 2.0 (#7)

notwaldorf commented on polymer.json:
```diff
@@ -0,0 +1,17 @@
+{
+  "entrypoint": "index.html",
```
Why is this file even needed?

notwaldorf replied:
Ah, please don't add a service worker. This is a standalone element, and a service worker is a user feature, not an element feature.

mikeperret replied:
Not needed - honestly I thought it was needed to run "polymer serve" but it is not :-)

notwaldorf commented on service-worker.js:
```diff
@@ -0,0 +1,10 @@
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+console.log('Service worker disabled for development, will be generated at build time.');
```
Please remove, as mentioned.

notwaldorf commented on test/basic.html:
```diff
@@ -101,7 +100,7 @@
             expect(rain.importedTwemoji).to.be.eql(true);
             expect(rain.active).to.be.eql(true);
-            done();
+            //done(); (thrown error - why?)
```
Uncomment and add a `done` function to the test instead

notwaldorf commented on test/index.html:
```diff
@@ -2,8 +2,8 @@
   <meta charset="utf-8">
-  <script src="../../webcomponentsjs/webcomponents-lite.js"></script>
-  <script src="../../web-component-tester/browser.js"></script>
+  <script src="../bower_components/webcomponentsjs/webcomponents-lite.js"></script>
```
Not needed (see above)

notwaldorf commented on test/basic.html:
```diff
@@ -3,11 +3,10 @@
-  <script src="../../test-fixture/test-fixture-mocha.js"></script>
-  <link rel="import" href="../../polymer/polymer.html">
-  <link rel="import" href="../../test-fixture/test-fixture.html">
+  <script src="../bower_components/webcomponentsjs/webcomponents-lite.js"></script>
```
Not needed (see above)

notwaldorf commented on demo/index.html:
```diff
@@ -7,8 +7,7 @@
   <link href='http://fonts.googleapis.com/css?family=Lato:300,400' rel='stylesheet' type='text/css'>
-  <script src="../../webcomponentsjs/webcomponents-lite.js"></script>
-  <link rel="import" href="../../polymer/polymer.html">
+  <script src="../bower_components/webcomponentsjs/webcomponents-lite.js"></script>
```
This isn't needed -- this element is meant to be a standalone element, so you should test it using `polymer serve`. See `github.com/polymerelements/paper-button` for another element sample.

notwaldorf reviewed (changes requested):
Thanks for the PR! There's a bunch of changes that aren't needed, though, since this is a standalone element (not a shipped demo)

mikeperret committed:
Fixed 2.0 conversion issues (#7)

mikeperret commented:
Thank you for your feedback and sorry for my lack of understanding - I should have gotten a lot more suspicious about changing all these paths in the first place.

**Changes**
- Removed polymer.json
- Removed service-worker.js
- Revoked changes of paths in `<imports>/<scripts>`-Tags in all .html files
- Removed a unnecessary console.log I added in emoji-rain.html
- Removed the speed-attribute I added in emoji-rain.html (because I think I should refactor it a bit and add it inside another PR)
- Fixed test/basic.html `done();` problem

notwaldorf commented:
Super late, but yay! Thanks for the PR!
